### PR TITLE
Danpf/uninitialized copy fix

### PR DESCRIPTION
**Pull request by danpf**
Added docs for debugging + a fix for unitialized_copy bindings.

### CODE REVIEW

File: documentation/debugging.rst [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+Here are some helpful tips that might help you on your way to making bindings.
+
+
+---------------
+inconsistencies
+---------------
+
+Binder moves down your ``all_includes_file`` sequentially, sometimes you might end up with errors,
+and sometimes you might not.  This is almost always caused by your ``all_includes_file`` being
+different between runs.  The order shouldn't really matter, but nail it down to at least be
```
Comment (lyskov) [inline documentation/debugging.rst]: Maybe 'The order' → 'The order of includes' for clarity?

File: documentation/debugging.rst [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+--------------
+Build failures
+--------------
+
+Sometimes you might get an error when building your compiled module.  binder has trouble  
+when standard libraries include eachother.  We have already mapped many of these
+interactions in ``types.cc``, but we still might be missing some internal knowledge about
+all of the c++ std library modules out there.  An error like this almost always
+manifests itself with an abundance of errors complaining about ``std::bits`` or ``bits``.
```
Comment (lyskov) [inline documentation/debugging.rst]: Hmm... unless i am mistaken code is not in `std::bits` but rather in directory `bits/bits_<some_std_include>`

File: documentation/debugging.rst [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    In file included from std/complex.cpp:1:0:
+    /usr/include/c++/7/bits/stl_construct.h: In function ‘void std::_Destroy(_ForwardIterator, _ForwardIterator)’:
+    **bunch of garbage below here**
+
+1.  Rebuild bindings, but now add the flag ``--annotate-includes`` which will give us much
+    more information than we would normally get
+
+2.  Since we know that ``bits`` is bad for binder, we can just grep for bits in the directory where we
+    generate and compile the bindings.  In the ``examples`` section, this is the same as the directory
+    ``cmake_bindings``.
```
Comment (lyskov) [inline documentation/debugging.rst]: Probably a bit of info here would be better approach. I suggest re-wording this along this lines:
... header in bits/ is inner std implementation header that should not be included directly. So if our generated code ends up with such include files it is a sign that Binder `type.cpp` file might need updating by adding additional mapping between std::<some-type> and appropriate std header file.

**Comment by kostyfisik** [comment]
Any chance that this PR will be accepted? (So I can continue with my PR #55 )

**Comment by everyday847** [comment]
I anticipate it will be if you make some of the suggested changes!

**Comment by lyskov** [comment]
@kostyfisik - certainly! Please address a minor changes above and we can merge this. Thanks,

**Comment by lyskov** [comment]
@danpf looks like this got abandon. Should i close it for now or do you plan to finish it someday? Thanks,

**Review by lyskov — DISMISSED** [review]
@danpf comments for this PR was addressed in ##139 so will merge this as is. - thank you for putting this together!

**Review by lyskov — APPROVED** [review]

**Pull request merged**